src/selector.jl

```diff
@@ -149,7 +152,7 @@ relate(::Forward, dim, val) = val
 relate(::Reverse, dim, val) = lastindex(dim) .- val .+ 1
 
 # Selector indexing without dim wrappers. Must be in the right order!
-Base.@propagate_inbounds Base.getindex(a::AbstractArray, I::Vararg{Selector}) =
+Base.@propagate_inbounds Base.getindex(a::AbDimArray, I::Vararg{Union{Selector, StandardIndices}}) =
     getindex(a, sel2indices(a, I)...)
 Base.@propagate_inbounds Base.setindex!(a::AbstractArray, x, I::Vararg{Selector}) =
     setindex!(a, x, sel2indices(a, I)...)
```

Comment:
We should also make this change for `setindex!` and `view`. You could make a `SelectorOrStandard` constant for the type.

Reply:
> We should also make this change for setindex! and view.

That's the plan!

Why were all these methods defined for `AbstractArray` and not `AbstractDimArray`? Modify that doesn't break any tests, but I wasn't sure if there was some untested behaviour I should be aware of here.

Reply:
Great!

I originally had the idea that you don't have to inherit from AbstractDimensionalArray to use a lot of this package. Just a `dims` and `rebuild` method and dimension indexing gets you a long way. 

It might not end up making sense. My goal was to make everything as widely extensible as possible, as the opposite of AxisArrays that really isn't extensible at all. I probably went too far.

Reply:
I'm not sure it makes sense for this method to apply to normal arrays, but it would be cool if this could dispatch on some meta-package's trait for labelled arrays. I had to change it from `AbstractArray` here since it was causing dispatch errors to be that general.

Reply:
Just had a thought about this and the use of `sel2indices` in the tests.

Why not assign these methods to `to_indices(::DimensionalArray, ...)`?